src/traefik.py

```diff
@@ -0,0 +1,574 @@
+
+    # wokeignore:rule=master
+    # ref: https://doc.traefik.io/traefik/master/observability/tracing/opentelemetry/
+    def config_tracing(self, endpoint: str, grpc: bool):
+        """Push yaml config with opentelemetry configuration."""
+        config = yaml.safe_dump(
+            {
+                "tracing": {
+                    "openTelemetry": {
+                        "address": endpoint,
+                        **({"grpc": {}} if grpc else {}),
+                        # todo: we have an option to use CA or to use CERT+KEY (available with mtls) authentication.
+                        #  when we have mTLS, consider this again.
+                        **({"ca": CA_CERT_PATH} if self._tls_enabled else {"insecure": True}),
+                    }
+                }
+            }
+        )
+        logger.debug(f"dumping tracing config to {DYNAMIC_TRACING_PATH}")
+
+        self._container.push(DYNAMIC_TRACING_PATH, config, make_dirs=True)
+
+    def get_per_unit_http_config(
+        self,
+        *,
+        prefix: str,
+        scheme: str = "http",
+        host: str,
+        port: int,
+        strip_prefix: bool,
+        redirect_https: bool,
+        external_host: str,
+    ) -> dict:
+        """Generate a config dict for IngressPerUnit."""
+        lb_servers = [{"url": f"{scheme}://{host}:{port}"}]
+        return self._generate_config_block(
+            prefix=prefix,
+            lb_servers=lb_servers,
+            scheme=scheme,
+            strip_prefix=strip_prefix,
+            redirect_https=redirect_https,
+            external_host=external_host,
+        )
+
+    def get_per_app_http_config(
+        self,
+        *,
+        prefix: str,
+        scheme: str = "http",
+        hosts: List[str],
+        port: int,
+        strip_prefix: bool,
+        redirect_https: bool,
+        external_host: str,
+    ) -> dict:
+        """Generate a config dict for Ingress(PerApp)."""
+        lb_servers = [{"url": f"{scheme}://{host}:{port}"} for host in hosts]
+        return self._generate_config_block(
+            prefix=prefix,
+            lb_servers=lb_servers,
+            scheme=scheme,
+            strip_prefix=strip_prefix,
+            redirect_https=redirect_https,
+            external_host=external_host,
+        )
+
+    def get_per_leader_http_config(
+        self,
+        *,
+        prefix: str,
+        scheme: str = "http",
+        host: str,
+        port: int,
+        strip_prefix: bool,
+        redirect_https: bool,
+        external_host: str,
+    ) -> dict:
+        """Generate a config dict for Ingress v1 (PerLeader)."""
+        lb_servers = [{"url": f"http://{host}:{port}"}]
+        return self._generate_config_block(
+            prefix=prefix,
+            lb_servers=lb_servers,
+            scheme=scheme,
+            strip_prefix=strip_prefix,
+            redirect_https=redirect_https,
+            external_host=external_host,
+        )
+
+    def _generate_config_block(
```

Comment:
This and other methods do config + workload concerns.
Totally fine. Just pointing out if you intended otherwise.

Reply:
mmmh just to make sure we're on the same page, my idea was that the Traefik class should handle the workload configuration (push/pull/remove config yamls), and (mediate, or initiate) workload operations such as starting/restarting.

Are you saying something else?

Reply:
Ok, will try this out in a follow-up PR